StackOverflowError ClassWalker

**Describe the bug**
StackOverflowError occurs when walking entity class to map content properties.

**Additional context**
Partial stack trace is:

`Caused by: java.lang.StackOverflowError at java.base/java.lang.reflect.Field.getAnnotation(Field.java:1142) at java.base/java.lang.reflect.AnnotatedElement.isAnnotationPresent(AnnotatedElement.java:274) at java.base/java.lang.reflect.AccessibleObject.isAnnotationPresent(AccessibleObject.java:517) at org.springframework.content.commons.mappingcontext.ContentPropertyBuilderVisitor.visitField(ContentPropertyBuilderVisitor.java:92) at org.springframework.content.commons.mappingcontext.ClassWalker.accept(ClassWalker.java:24)`

--- a/spring-content-commons/src/main/java/org/springframework/content/commons/mappingcontext/ContentPropertyBuilderVisitor.java
+++ b/spring-content-commons/src/main/java/org/springframework/content/commons/mappingcontext/ContentPropertyBuilderVisitor.java
@@ -163,6 +163,7 @@ public class ContentPropertyBuilderVisitor {
             }
         } else if (f.getType().isPrimitive() == false &&
                    f.getType().equals(String.class) == false &&
+                   f.getType().isEnum() == false &&
                    ContentPropertyUtils.isWrapperType(f.getType()) == false &&
                    ContentPropertyUtils.isRelationshipField(f) == false) {
 

--- a/spring-content-commons/src/test/java/org/springframework/content/commons/mappingcontext/ClassWalkerTest.java
+++ b/spring-content-commons/src/test/java/org/springframework/content/commons/mappingcontext/ClassWalkerTest.java
@@ -150,6 +150,8 @@ public class ClassWalkerTest {
 
     public static class TestClass {
 
+        private TestEnum enums;
+
         private TestClass circlularRef;
 
         private String str;
@@ -217,5 +219,9 @@ public class ClassWalkerTest {
     }
 
     public static class TestClass4 extends TestSubSubClass {
+    }
+
+    public static enum TestEnum {
+        A, B, C, D
     }
 }

